aces/Loris
# [API] Fix PUT request for Candidates \visit endpoint and integration test

## Brief summary of changes

The PUT request should never allow to edit a candidate from a `Site `where the User has no affiliation with. In this case, the test User has only affiliation with Data Coordinating Center (DCC). It is however possible to do so if the new `Site` is DCC.

For example, in raisinbread, candidate 300001 (`https://<hostname>.loris.ca/api/v0.0.4-dev/candidates/300001`) is from site `Montreal`. The test user should not be able to modify this candidate, because the candidate is from Montreal, but is able to.

Also adds automatic the integration tests to test the changes suggested.

#### Testing instructions

The testing is made automatic.

To test manually: 

1. Before the test, make sure the User that will be used don't have affiliation with Montreal.

2. First get the API token using: `curl https://<hostname>/api/v0.0.3/login -d '{"username": "<username>", "password": "<password>"}'`

3. In the terminal, enter `curl -X PUT -H "Authorization: Bearer $token" https://<hostname>/api/v0.0.4-dev/candidates/300001/V3 -d '{"CandID":"300001","Visit":"V3","Site":"Montreal","Battery":"Low Yeast","Project":"Pumpernickel"}'`


#### Link(s) to related issue(s)

* Resolves #7106
